zaza/openstack/charm_tests/rabbitmq_server/tests.py

```diff
@@ -461,6 +461,61 @@ def check_units(units):
 
         logging.info('OK')
 
+    def test_policies(self):
+        """Test config policies."""
+        app_config = zaza.model.get_application_config(self.application_name)
+        policies_config = json.loads(
+            app_config.get('policies')
+            .get('default')
+            .replace("\n", "")
+            .replace(" ", "")
+        )
+
+        # check policies implemented in the clusterer match with config
+        self._check_policies(policies_config)
+
+        # add a new policy
+        new_policy = {
+            'vhost': '/',
+            'name': 'test',
+            'pattern': 'foo',
+            'type': 'ttl'
+        }
+        policies_config_changed = policies_config + [new_policy]
+        zaza.model.set_application_config(
+            'rabbitmq-server',
+            {'policies': json.dumps(policies_config_changed)}
+        )
+        zaza.model.block_until_all_units_idle()
+
+        # check again after adding a new policy
+        self._check_policies(policies_config_changed)
+
+    @tenacity.retry(
+        wait=tenacity.wait_exponential(max=60),
+        stop=tenacity.stop_after_attempt(8)
+    )
+    def _check_policies(self, policies_config):
+        """Compare cluster policies with the juju config.
+
+        :param policies_config: Policies from juju config
+        :type policies_config: [Dict]
```

Comment:
More correct type is `List[Dict[str, <fill in type>]]`

If possible, be as concrete as possible as some future version of this will check the types.  Also editors using pyright will get linting and type checking.